packages/core/src/components/grading-instructions/GradingInstructions.tsx

```diff
@@ -109,30 +113,32 @@ const GradingInstructions: React.FunctionComponent<CommonExamProps & AnnotationP
       onClickAnnotation={onClickAnnotation}
       onSaveAnnotation={onSaveAnnotation}
     >
-      <I18nextProvider i18n={i18n}>
-        <main className="e-exam e-grading-instructions" lang={subjectLanguage}>
-          <React.StrictMode />
-          <SectionElement aria-labelledby={examTitleId}>
-            <DocumentTitle id={examTitleId}>
-              <GradingInstructionsPageTitle /> {examTitle && renderChildNodes(examTitle)}
-            </DocumentTitle>
-            {date && (
-              <p>
-                <strong>{dateTimeFormatter.format(date)}</strong>
-              </p>
-            )}
-            {examGradingInstruction && (
-              <ExamGradingInstruction {...{ element: examGradingInstruction, renderChildNodes }} />
-            )}
-            {tableOfContents && (
-              <div className="main-toc-container">
-                <TableOfContents {...{ element: tableOfContents, renderChildNodes }} />
-              </div>
-            )}
-          </SectionElement>
-          {renderChildNodes(root)}
-        </main>
-      </I18nextProvider>
+      <GradingInstructionProvider {...{ editable, onContentChange, saveScreenshot }}>
```

Comment:
jotenkin nyt pistää silmään, että provider ei hallitse omaa tilaansa vaan se tulee propseina ja mietin onko se ongelma 🤔 tai en ole tälläistä ratkaisua ehkä näin vaan nähnyt.